crates/uv-pep508/src/marker/tree.rs

```diff
@@ -716,6 +720,23 @@ impl MarkerTree {
         self.or(consequent);
     }
 
+    /// Simplify the marker, namely, by detecting known impossible conditions.
+    ///
+    /// For example, while the marker specification and grammar do not _forbid_ it, we know that
+    /// both `sys_platform == 'win32'` and `platform_system == 'Darwin'` will never true at the
+    /// same time.
+    ///
+    /// This method thus encodes assumptions about the environment that are not guaranteed by the
+    /// PEP 508 specification alone.
+    fn simplify(&mut self) {
+        if !self.0.is_false() {
+            let mutex = &*MUTUAL_EXCLUSIONS;
+            if INTERNER.lock().is_disjoint(self.0, mutex.0) {
```

Comment:
Took me a while to understand that this is `is_disjoint` and not `!is_disjoint` because `MUTUAL_EXCLUSIONS` is the negated exclusions, so it's the valid space.

![image](https://github.com/user-attachments/assets/28d16937-db91-4603-b8a2-dafce9ea9b8c)

Plus when reading the code you need to consider we call this every time while building the marker, so when only B from `A or B` lies in the excluded area, we still get A because B was converted to FALSE upon construction.

Reply:
Should I be changing something in response to this comment? :)

Reply:
Could you add an inline comment like:

> `MUTUAL_EXCLUSIONS` is the possible space, if the markers are `is_disjoint` with it, they are solely in the impossible space.

Reply:
(I'll leave the smart constructor review to andrew/ibraheem)